packages/shopify-codemod/transforms/split-return-assignments.js

```diff
@@ -0,0 +1,112 @@
+import {matchLast} from './utils';
+
+export default function splitReturnAssignments({source}, {jscodeshift: j}, {printOptions = {}}) {
+  const sourceAST = j(source);
+  sourceAST
+    .find(j.Function, {
+      body: {
+        body: matchLast(j.ReturnStatement, {
+          type: 'ReturnStatement',
+          argument: {
+            type: 'AssignmentExpression',
+          },
+        }),
+      },
+    })
+    .forEach(({node: {body: {body}}}) => {
+      const {left, right} = body[body.length - 1].argument;
+      const returnLine = body[body.length - 1];
+      if (returnLine.argument.left) {
+        delete body[body.length - 1];
+        body.push({
+          type: 'ExpressionStatement',
+          expression: {
+            type: 'AssignmentExpression',
+            operator: '=',
+            left,
+            right,
+          },
+        });
+
+        if (returnLine.argument.left.name) {
```

Comment:
If you find yourself reaching deep into a nested property multiple times, you probably want to pull that property up as its own variable in scope/